### ISSUE TITLE
CSVRecordReader does not parse correctly with pipe | as delimiter

### ISSUE THREAD

**Issue by mdebeer** [deeplearning4j/DataVec#414]
#### Issue Description

Here is a cut of my data file:
```
17|0|0|2016-03-20 00:00:00.0000000|1|13|1|0||
17|0|0|2016-03-27 00:00:00.0000000|1|95|1|0||
17|0|0|2016-04-03 00:00:00.0000000|1|15|1|0||
17|0|0|2016-04-10 00:00:00.0000000|1|25|1|0||
```

Using CSVRecordReader:
```
RecordReader rr = new CSVRecordReader(1, "|");
rr.initialize(new FileSplit(new File(inputPath));
```

Printing rr.next() gives:
`1, 7, |, 0, |, 0, |, 2, 0, 1, 6, -, 0, 3, -, 2, 0,  , 0, 0, :, 0, 0, :, 0, 0, ., 0, 0, 0, 0, 0, 0, 0, |, 1, |, 1, 3, |, 1, |, 0, |, |, ]
`
So it is splitting on each character, instead of dividing into the correct columns.

When I change my file's delimiter to comma or semicolon it appears the read properly:
`[17, 0, 0, 2016-03-20 00:00:00.0000000, 1, 13, 1, 0, , ]`

#### Version Information

* Deeplearning4j version
Believe that it's latest -- according to Version file: 0.6.1-SNAPSHOT

**Comment by huitseeker** [comment]
Thank you very much for this detailed, reproducible report. I've added a test case for it in https://github.com/deeplearning4j/DataVec/pull/415 — it appears this issue is fixed by 0.9.1, probably by cc2f6f440 and I'm afraid you'll have to upgrade to get the bugfix :)

**Comment by mdebeer** [comment]
Aah, thank you for your contribution! Fortunately there is also a simple fix :)

> Alex Kozlov @alexvk 17:16
> It's splitting on regex, try using '\|' instead

👍
